lib/matplotlib/backend_tools.py

```diff
@@ -1018,6 +1021,71 @@ def _mouse_move(self, event):
         self.toolmanager.canvas.draw_idle()
 
 
+class HelpTool(ToolToggleBase):
+    description = 'Print tool list, shortcuts and description'
+    default_keymap = rcParams['keymap.help']
+    image = 'help.png'
+
+    def __init__(self, *args):
+        ToolToggleBase.__init__(self, *args)
+        self.text_axes = None
+
+    def enable(self, *args):
+        # Using custom axes label to prevent reuse of old axes
+        # https://github.com/matplotlib/matplotlib/issues/9024
+        self.text_axes = self.figure.add_axes((0, 0, 1, 1),
+                                              label='help_tool_axes')
+        table = Table(self.text_axes, bbox=[0, 0, 1, 1])
+        table.edges = 'B'
+        self.text_axes.add_table(table)
+        chars_in_width = self._find_chars_in_width(table.FONTSIZE)
+        col_chars_width = int(chars_in_width / 6)
+        content = self._get_content(1 * col_chars_width - 2,
+                                    2 * col_chars_width - 2,
+                                    3 * col_chars_width - 2)
+        for i, v in enumerate(content):
+            h = v[0]
+            table.add_cell(i, 0, text=v[1], width=1, height=h, loc='left',
+                           fontproperties='monospace')
+            table.add_cell(i, 1, text=v[2], width=2, height=h, loc='left',
+                           fontproperties='monospace')
+            table.add_cell(i, 2, text=v[3], width=3, height=h, loc='left',
+                           fontproperties='monospace')
+        table.auto_set_font_size(True)
+        self.figure.canvas.draw_idle()
+
+    def _find_chars_in_width(self, fontsize):
+        """Number of characters in figure width approx"""
+        # https://web.archive.org/web/20010717031241/plainlanguagenetwork.org/type/utbo211.htmhttps://web.archive.org/web/20010717031241/plainlanguagenetwork.org/type/utbo211.htm
+        # Approximately width = 60% of height for monospace fonts
+        charwidth = 0.6 * fontsize / 72.0 * self.figure.dpi
+        figwidth = self.figure.get_figwidth() * self.figure.dpi
+        return figwidth / charwidth
+
+    def disable(self, *args):
+        self.text_axes.remove()
+        self.figure.canvas.draw_idle()
```

Comment:
Does this mean that after having pressed that button 10 times, you have 10 unused `Axes` in memory? Should it maybe rather reuse an existing axes, or comletely delete the unused axes?

Reply:
when creating the axes we specify the label, so if I understand it correctly it should reuse it if available

Reply:
That behavior (of reusing axes) is deprecated.

Reply:
But's it's not available, because you removed it from the figure. And if you didn't remove it, it would cause a warning, telling you that creating an instance with the exact same arguments will return a new instance in the future. 

Simple test case:
```
fig = plt.figure()
ax1 = fig.add_axes((0, 0, 1, 1), label='help_tool_axes')
ax1.remove()
ax2 = fig.add_axes((0, 0, 1, 1), label='help_tool_axes')
print(ax1==ax2)    # This will print False.
```

I think it would indeed be a good idea to reuse the axes, but I currently would not know how to do that. I.e. how would one add an axes to a figure which has previously been removed? (`fig.add_axes(ax)` does not seem to work.)

Reply:
Create it once, call `set_visible(True)`, call `set_visible(False)`.

Reply:
Done, now it creates it once, set the visibility to change.
I clear the axes before filling it because the tools are dynamic, so the content might change between different calls

Reply:
Setting the visibility lets the axes stay in `fig.axes`. This might change any program's complete logic depending on whether or not you are using the ToolManager to show the figure and whether or not you have previously toggled the help or not. 

Simple example:
```
for ax in fig.axes:
    # do something.
```